cardano-testnet/src/Testnet/EpochStateProcessing.hs

```diff
@@ -75,3 +83,35 @@ maybeExtractGovernanceActionIndex sbe txid (AnyNewEpochState actualEra newEpochS
     compareWithTxId (Api.TxId ti1) Nothing (GovActionId (TxId ti2) (L.GovActionIx gai)) _
       | ti1 == L.extractHash ti2 = Just gai
     compareWithTxId _ x _ _ = x
+
+-- | Watch the epoch state view until the guard function returns @Just@ or the timeout is reached.
+-- | Wait for at least `minWait` epochs and at most `maxWait` epochs.
+-- | The function will return the result of the guard function if it is met, otherwise it will return @Nothing@.
+watchEpochStateView
+  :: forall m a. (MonadIO m, MonadTest m, MonadAssertion m)
+  => EpochStateView -- ^ The info to access the epoch state
+  -> (AnyNewEpochState -> m (Maybe a)) -- ^ The guard function (@Just@ if the condition is met, @Nothing@ otherwise)
+  -> Word64 -- ^ The minimum number of epochs to wait
+  -> Word64 -- ^ The maximum number of epochs to wait
+  -> m (Maybe a)
+watchEpochStateView epochStateView f minWait maxWait = do
+  AnyNewEpochState _ newEpochState <- getEpochState epochStateView
+  let (EpochNo currentEpoch) = L.nesEL newEpochState
+  go (EpochNo $ currentEpoch + minWait) (EpochNo $ currentEpoch + maxWait)
+    where
+      go :: EpochNo -> EpochNo -> m (Maybe a)
+      go (EpochNo startEpoch) (EpochNo timeout) = do
```

Comment:
```suggestion
      go :: Word64 -> Word64 -> m (Maybe a)
      go startEpoch timeout = do
```
It's an internal function, so I don't think there's any benefit of using `EpochNo` here.

Reply:
I thought the same thing, but because they are computed slots (relative to current slot) they are not in the main function's closure, so they are not accessible by the auxiliary function directly

Reply:
Argh, I misunderstood and skipped it originally. Well, it doesn't really hurt anyway and it is easy to confuse it with an `EpochInterval`. I am going to leave it there for now